test-utils/src/lib.rs

```diff
@@ -39,7 +39,21 @@ pub fn decode_from_hex<D: serialization::DecodeAll>(to_decode: &str) -> D {
         .expect("The decoding succeeded")
 }
 
-pub fn random_string<R: SampleRange<usize>>(rng: &mut impl Rng, range_len: R) -> String {
+pub fn get_random_non_ascii_alphanumeric_byte(rng: &mut impl Rng) -> u8 {
+    for _ in 0..1000 {
+        let random_byte = rng.gen::<u8>();
+        if !random_byte.is_ascii_alphanumeric() {
+            return random_byte;
+        }
+    }
+    // it's approximately 0.5^1000 that this panics
```

Comment:
It's actually much less... because half of u8 range is printable characters, not necessarily alphanumeric.